lib/common_test/src/unix_telnet.erl

```diff
@@ -100,6 +102,28 @@ Setup Telnet connection to a Unix host.
 For `target_name()`, see `m:ct`. For `handle()`, see `m:ct_telnet`.
 """.
 -doc(#{since => <<"OTP 18.3.3">>}).
+-spec connect(ConnName, Ip, Port, Timeout, KeepAlive, TCPNoDelay, Extra) ->
+          {'ok', Handle} | {'error', Reason}
+              when ConnName :: ct:target_name(),
+                   Ip :: inet:socket_address() | inet:hostname(),
+                   Port :: inet:port_number(),
+                   Timeout :: timeout(),
+                   KeepAlive :: boolean(),
+                   TCPNoDelay :: boolean(),
+                   Extra :: {Username, Password} | KeyOrName,
+                   Username :: iodata(),
+                   Password :: iodata(),
+                   KeyOrName :: atom(),
+                   Handle :: ct_telnet:handle(),
+                   Reason :: {'no_username', KeyOrName} | {'no_password', KeyOrName}
+                           | {'timeout' | inet:posix(), ct_telnet:handle()}
+                           | {'error', 'bad_pattern', Pattern}
+                           | {'error', 'bad_pattern', Pattern, CompileError}
+                           | {'error', 'timeout'} | {'error', term()},
```

Comment:
@IngelaAndin @u3s - shouldn't the code here be fixed to not produce nested error tuples?

Reply:
Will write another comment, listing all parts I think are incorrect.